libp2p/multiaddress.nim

```diff
@@ -1005,7 +1005,7 @@ proc `$`*(pat: MaPattern): string =
 proc write*(pb: var ProtoBuffer, field: int, value: MultiAddress) {.inline.} =
   write(pb, field, value.data.buffer)
 
-proc getField*(pb: var ProtoBuffer, field: int,
+proc getField*(pb: ProtoBuffer, field: int,
```

Comment:
Why remove `var`,  no reason to pass by copy?

Reply:
All getField had their `var` remove a while ago, this was an old remainder (which was incompatible with the new getField without var)

Reply:
hmm, I can't recall the reason why it was removed? This would incur in potential copying overhead without any clear benefits?

Reply:
> pass by copy

per the same comment in the other branch, nim passes by reference, when that is appropriate, regardless of `var`/`let`

Reply:
Isn't that an artifact of the optimizer and it wont work in certain cases (not to mention that it's platform dependent and hard to predict). `var` makes it explicit and outside of the optimizers reach? Copying thousands of bytes doesn't seem to be very efficient?

Reply:
`var` makes explicit that the instance is mutable -it's unrelated to references - that a reference is used for either of `let` or `var` is an implementation detail (and one that nim _generally_ gets right, when it comes to passing variables). 

When the function doesn't need to mutate its parameter, we don't use `var`: https://status-im.github.io/nim-style-guide/language.vardecl.html - in part because `var` is infectious: callers must now also use `var`, where they could have used `let` or `const`.

Reply:
I agree that it forces weird semantics on the caller, but it's also the only way of passing a stack object "by ref", whats the alternative?

Reply:
the alternative is to trust the compiler to do the right thing, which is what we do pretty much everywhere.